### PR TITLE
bugfix

### PR DESCRIPTION
**Pull request by prokopsimek**
Fixes: https://github.com/refinery/refinerycms-blog/issues/367
Maybe fixes: https://github.com/refinery/refinerycms-blog/issues/361 https://github.com/refinery/refinerycms-blog/pull/363 https://github.com/refinery/refinerycms-blog/issues/365 https://github.com/refinery/refinerycms-blog/issues/342

HOW TO FIX older posts:
1) pull this fix
2) Start rails console (rails c)
3) In rails console run command:  ::Refinery::Blog::Post.find_each(&:save)

And it should be ok. It is for me...

### CODE REVIEW

File: app/controllers/refinery/blog/admin/posts_controller.rb [hunk 1]
```diff
@@ -86,7 +86,8 @@ def post_params
       protected
 
         def find_post
-          @post = Refinery::Blog::Post.find_by_slug_or_id(params[:id])
+          @post = Refinery::Blog::Post.find_by_slug(params[:id]) if params[:id].friendly_id?
+          @post = Refinery::Blog::Post.find(params[:id]) if @post.nil?
```
Comment (parndt) [inline app/controllers/refinery/blog/admin/posts_controller.rb]: I think that ultimately this should just be:

```ruby
@post = Refinery::Blog::Post.friendly.find(params[:id])
```

FriendlyId handles this internally.

Comment (prokopsimek) [inline app/controllers/refinery/blog/admin/posts_controller.rb]: Hi, I preffer friendly_id initializer...
Dne 20.2.2015 5:21 "Philip Arndt" <notifications@github.com> napsal(a):

Comment (prokopsimek) [inline app/controllers/refinery/blog/admin/posts_controller.rb]: To use "config.use :finders"
Dne 20.2.2015 8:54 "Prokop Simek" <gilotka@gmail.com> napsal(a):

Comment (simi) [inline app/controllers/refinery/blog/admin/posts_controller.rb]: @prokopsimek does @parndt 's version work for you? It looks like the best idea how to solve this problel.

Comment (prokopsimek) [inline app/controllers/refinery/blog/admin/posts_controller.rb]: yes

**Comment by parndt** [comment]
This is now irrelevant since [this was merged](https://github.com/refinery/refinerycms-blog/compare/bb0d3b0...4efb627) with the same fix.

**Comment by simi** [comment]
splendid

2015-05-14 0:28 GMT+02:00 Philip Arndt <notifications@github.com>:

**Pull request closed without merging**